### PR TITLE
[Refactor and Cleanup] Task Code

### PR DESCRIPTION
**Pull request by Akkadius**
**All commits will be squashed into one on merge**

This PR mainly achieves the following

* High level formatting, renaming, refactoring to make things easier to understand without aggressively refactoring data structures and internals
* Replaces a majority of database persistence logic with repositories which makes the code far easier to reason about
* Private class members prefixed with `m_`
* All function params refactored with `snake_case`
* Function variables renamed to give something easier to understand rather than cryptic names and acronyms like `i` or `j` or `c` or `ctm`
* 95% of task function scoped variables renamed with `snake_case`
* Added pointer aliases to make it easier to understand the code where there is heavy nested array traversal
* Indexes are at least appropriately named with something easier to understand rather than `i` or `j`
* Most log statements have been reformatted and ported to `LogTasks` or `LogTasksDetail` aliases

**Warning Cleanup**

* World wide task commands used to throw compiler errors because of improper casting; those have been eliminated in this PR cc @KinglyKrab 

**Repository Adjustments**

* cmakelists updated with all recent repositories
* FindOne in each repository was missing being passed a database pointer which was adjusted and all repositories regenerated

**Repository Usage**

Lots of persistence layer usage has been migrated to use the generated repositories. It is much cleaner to use and work with

Example(s)

```cpp
	auto proximities = ProximitiesRepository::GetWhere(
		content_db,
		fmt::format("zoneid = {} ORDER BY `zoneid` ASC", zone_id)
	);

	CharacterActivitiesRepository::DeleteWhere(
		database,
		fmt::format("charid = {} AND taskid = {}", character_id, task_id)
	);

	CharacterTasksRepository::DeleteWhere(
		database,
		fmt::format("charid = {} AND taskid = {} AND type = {}", character_id, task_id, static_cast<int>(task_type))
	);
```

**Task Code Relocation**

Before, task code existed entirely in a very long tasks.cpp and tasks.h, making incremental builds painful anytime anything was touched and also had several classes maintained in one super file

Code is now split out; tasks.cpp contains only a few client members while tasks.h contains any type information that isn't already specific to the class/file it is used within

Also makes the code easier to reason about with code organized in their own places instead of all crammed into a single file

```
zone/task_client_state.cpp
zone/task_client_state.h
zone/task_goal_list_manager.cpp
zone/task_goal_list_manager.h
zone/task_manager.cpp
zone/task_manager.h
zone/task_proximity_manager.cpp
zone/task_proximity_manager.h
zone/tasks.cpp
zone/tasks.h
```

**GM Command Menu**

Some quality of life changes have been made to the in game GM command menu with the introduction of saylinks, new commands etc.

* #task commands will now work on any targeted client, not just the GM who invoked them
* Added #task assign <task_id> which will assign a task to the targeted client
* #task show now displays saylink shortcuts that will update an activity which helps with rapid testing

image
image
image

**Testing Performed**

After refactoring, the following tests were performed to ensure things still work as expected

* Getting a task
* Deleting a task
* Making sure task still shows up after zoning
* Making sure task activity state remains the same after zoning
* Completion, deletion etc.
* Updating task activities, new steps etc

### CODE REVIEW

File: zone/task_client_state.cpp [hunk 1]
```diff
@@ -0,0 +1,2177 @@
...
+	// The activity_id is out of range
+	if (activity_id >= p_task_data->activity_count) {
+		return false;
+	}
+
+	LogTasks(
+		"[IsTaskActivityActive] (Update) task_id [{}] activity_id [{}] activity_state",
+		task_id,
+		activity_id,
+		info->activity[activity_id].activity_state
```
Comment (hgtw) [inline zone/task_client_state.cpp:1408]: Missing fmt braces here for the 3rd argument

File: zone/task_manager.cpp [hunk 2]
```diff
@@ -0,0 +1,1656 @@
...
+		// Task Activities MUST be numbered sequentially from 0. If not, log an error
+		// and set the task to nullptr. Subsequent activities for this task will raise
+		// ERR_NOTASK errors.
+		// Change to (activityID != (Tasks[taskID]->activity_count + 1)) to index from 1
+		if (activity_id != m_task_data[task_id]->activity_count) {
+			LogError(
+				"[TASKS]Activities for Task [{}] are not sequential starting at 0. Not loading task",
+				task_id,
+				activity_id
+			);
```
Comment (hgtw) [inline zone/task_manager.cpp:200]: Missing fmt string braces for `activity_id`

**Review by KinglyKrab — APPROVED** [review]

**Review by hgtw — CHANGES_REQUESTED** [review]

**Comment by KinglyKrab** [comment]
Looks good to me. Hg saw some missing things I didn't catch.

File: zone/task_client_state.cpp (hunk 1 above)
Comment (Akkadius) [reply zone/task_client_state.cpp:1408]: Good eye

**Review by hgtw — APPROVED** [review]

**Comment by Akkadius** [comment]
Was watching the superbowl and forgot to squash the commits on merge -_-

**Pull request merged**